src/view-strategy.js

```diff
@@ -253,3 +254,77 @@ export class InlineViewStrategy {
     return viewEngine.loadViewFactory(entry, compileInstruction, loadContext, target);
   }
 }
+
+@viewStrategy()
+export class StaticViewStrategy {
+
+  template: string | HTMLTemplateElement;
+  dependencies: Function[] | { (): Function[] | (Function | Promise<Function[] | Record<string, Function>>)[] };
+  factoryIsReady: boolean;
+  factory: ViewFactory;
+  onReady: Promsie<any>;
+
+  constructor(config: string | HTMLTemplateElement | { template: string; dependencies?: any } ) {
+    if (typeof config === 'string' || config instanceof HTMLTemplateElement) {
+      config = {
+        template: config
+      };
+    }
+    this.template = config.template;
+    this.dependencies = config.dependencies || [];
+    this.factoryIsReady = false;
+    this.onReady = null;
+  }
+
+  /**
+  * Loads a view factory.
+  * @param viewEngine The view engine to use during the load process.
+  * @param compileInstruction Additional instructions to use during compilation of the view.
+  * @param loadContext The loading context used for loading all resources and dependencies.
+  * @param target A class from which to extract metadata of additional resources to load.
+  * @return A promise for the view factory that is produced by this strategy.
+  */
+  loadViewFactory(viewEngine: ViewEngine, compileInstruction: IViewCompileInstruction, loadContext: IResourceLoadContext, target: any): Promise<IViewFactory> {
+    if (this.factoryIsReady) {
+      return Promise.resolve(this.factory);
+    }
+    let deps = this.dependencies;
+    deps = typeof deps === 'function' ? deps() : (deps ? Array.isArray(deps) ? deps : [deps] : []);
```

Comment:
Nit: no, thanks.
You did get the precedence of everything right, but that's too much trivia for real code.
I suggest _at least_ reordering the checks so that there is no ambiguity in the ternary operators results:
`isFunction ? deps() : isArray ? deps : deps || []`
That's still a bit cryptic to me, but at least there's no ambiguity in results. 
Note that it works because `isArray` simply returns false for anything that's not an array, including `null`, `undefined`.

Reply:
I think I have fixed this in the followup PR. Reading it again does feel terrible